app/src/main/java/org/astraea/partitioner/nodeLoadMetric/NodeMetrics.java

```diff
@@ -18,7 +17,7 @@
   private final JMXServiceURL serviceURL;
   private final MBeanClient mBeanClient;
   private final String nodeID;
-  private HashMap<String, Double> metricsValues;
+  private final NodeMetadata nodeMetadata;
```

Comment:
1. Could you change `NodeMetadata` to interface (i.e the `SafeMetadata` can be removed)?
2. `NodeMetrics` should implement the `NodeMetadata` interface.
3. `NodeMetrics` should be renamed to `NodeClient` as it has a member `MBeanClient`.
4. `NodeMetrics`  should implement the `AutoClosable` interface.